charts/cluster-infrastructure/templates/external-secrets.yml

```
@@ -0,0 +1,36 @@
+{{- if index .Values "external-secrets" "enabled" }}
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: external-secrets-{{ .Values.destination.name }}
+  namespace: argocd
+  labels:
+    application: external-secrets
+spec:
+  destination:
+    name: {{ .Values.destination.name }}
+    namespace: external-secrets
+  project: {{ .Values.destination.name }}
+  source:
+    repoURL: {{ index .Values "external-secrets" "image" "registry" }}
+    chart: {{ index .Values "external-secrets" "image" "repository" }}
+    targetRevision: {{ index .Values "external-secrets" "image" "tag" }}
```

I think this should work and is a little cleaner (we can backport in another PR)
```suggestion
    {{- with index .Values "external-secrets" "image" }}
    repoURL: {{ .registry }}
    chart: {{ .repository }}
    targetRevision: {{ .tag }}
    {{- end }
```